tests/unit/btree_stress_test.c

```diff
@@ -266,25 +277,138 @@ CTEST2(btree_stress, test_random_inserts_concurrent)
       (cache *)&data->cc, &data->dbtree_cfg, packed_root_addr, nkvs, data->hid);
    ASSERT_NOT_EQUAL(0, rc, "Invalid ranges in packed tree\n");
 
+   // Release memory allocated in this test case
+   for (uint64 i = 0; i < nthreads; i++) {
+      platform_free(data->hid, params[i].scratch);
+   }
+   platform_free(hid, params);
+}
+
+/*
+ * -------------------------------------------------------------------------
+ * Test case to exercise random inserts of large volumes of data, and then
+ * invoke some BTree-print methods, to verify that they are basically working.
+ * The initial work to setup the BTree and load some data is shared between
+ * this and the test_random_inserts_concurrent() sub-case.
+ * -------------------------------------------------------------------------
+ */
+CTEST2(btree_stress, test_btree_print_diags)
+{
+   int nkvs     = 1000000;
+   int nthreads = 8;
+
+   mini_allocator mini;
+
+   uint64 root_addr = btree_create(
+      (cache *)&data->cc, &data->dbtree_cfg, &mini, PAGE_TYPE_MEMTABLE);
+
+   platform_heap_id      hid    = platform_get_heap_id();
+   insert_thread_params *params = TYPED_ARRAY_ZALLOC(hid, params, nthreads);
+
+   load_thread_params(params,
+                      nthreads,
+                      data->hid,
+                      (cache *)&data->cc,
+                      &data->dbtree_cfg,
+                      &mini,
+                      root_addr,
+                      nkvs);
+
+
+   platform_status ret;
+   ret = do_n_thread_creates(
+      "insert thread", params, nthreads, data->ts, data->hid, insert_thread);
+   ASSERT_TRUE(SUCCESS(ret));
+
+   for (uint64 thread_no = 0; thread_no < nthreads; thread_no++) {
+      platform_thread_join(params[thread_no].thread);
+   }
+
+   uint64 packed_root_addr = pack_tests(
+      (cache *)&data->cc, &data->dbtree_cfg, data->hid, root_addr, nkvs);
+   if (0 < nkvs && !packed_root_addr) {
+      ASSERT_TRUE(FALSE, "Pack failed.\n");
+   }
+
    // Exercise print method to verify that it basically continues to work.
+   CTEST_LOG_INFO("\n**** btree_print_tree() on BTree root=%lu****\n",
+                  packed_root_addr);
```

Comment:
This is an existing invocation a BTree-print method, which has been working fine (previously, too).